src/main/java/com/deloitte/elrr/elrrconsolidate/consumer/ELRRMessageListener.java

```diff
@@ -86,60 +132,214 @@ private LearnerChange getLearnerChange(final String payload) {
         verbDisplay = verb.getDisplay().get("en-us");
       }
 
-      // Activity
-      Activity object = (Activity) statement.getObject();
+      // Object type
+      String objType = statement.getObject().getObjectType().name();
+      log.info("======> object type = " + objType);
 
-      // Activity name
-      String activityName = "";
-      String nameLangCode = "";
+      // If activity
+      if (objType.equalsIgnoreCase("ACTIVITY")) {
 
-      ActivityDefinition activityDefenition = object.getDefinition();
-      LangMap nameLangMap = activityDefenition.getName();
+        Activity activity = (Activity) statement.getObject();
 
-      if (nameLangMap != null) {
-        Set<String> nameLangCodes = nameLangMap.getLanguageCodes();
-        nameLangCode = nameLangCodes.iterator().next();
-        activityName = activityDefenition.getName().get(nameLangCode);
-      }
+        // Id
+        String id = "";
+        id = activity.getId();
```

Comment:
you should definitely not have a variable called `id`. There are like 10 `id`s in this scope.